### ISSUE TITLE
PyParsing 2.1 breaks SVG rendering library

### ISSUE THREAD

**Issue by corranwebster** [enthought/enable#224]
A change to the PyParsing library in 2.1.0 breaks the `enable.savage` SVG rendering library.  Versions <=2.0.3 seem to be fine.

**Comment by rkern** [comment]
Traceback? Failing example?

**Comment by corranwebster** [comment]
Sorry, it's all over the current Travis CI test runs, eg. https://travis-ci.org/enthought/enable/jobs/126210469 
```
======================================================================
ERROR: testEmptyURL (enable.savage.svg.tests.test_attributes.TestPaintValueURL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/enthought/enable/enable/savage/svg/tests/test_attributes.py", line 26, in testEmptyURL
    self.parser.parseString("url()").asList(),
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 1162, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 1028, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 2576, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 1028, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 2462, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 1032, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 2719, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 1054, in _parseNoCache
    tokens = fn( instring, tokensStart, retTokens )
  File "/home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pyparsing.py", line 792, in wrapper
    ret = func(*args[limit[0]:]) #~@$^*)+_(&%#!=-`~;:"[]{}
  File "/home/travis/build/enthought/enable/enable/savage/svg/attributes.py", line 17, in parsePossibleURL
    possibleURL, fallback = t[0]
ValueError: need more than 0 values to unpack
```

**Comment by corranwebster** [comment]
Temporary fix for CI is here #223

**Comment by rkern** [comment]
Works for me with `pyparsing 2.0.4`.

**Comment by rkern** [comment]
`2.1.0` shows the error, though.

**Comment by corranwebster** [comment]
My error from looking at the pyparsing website versions rather than at what's actually up on PyPI.

Will update issue appropriately.

**Comment by rkern** [comment]
It looks like the fix is to drop the `[0]` in that case. You can probably test on the length; if it's 1 item, then `t[0]`, if it's 2, then just `t`, if it's something else, then raise alarms.

**Comment by cournape** [comment]
Are we planning on releasing a `4..5.2` w/ a fix ? Or should I just implement what @rkern suggests in our package ?

**Comment by corranwebster** [comment]
I don't think we're going to do a 4.5.x release - might do a 4.6 release soon with Python 3 support.

This bug is only a problem in the savage SVG rendering subpackage and recent PyParsing, so it's not breaking Chaco and any users affected have a potential fix available.

Is there some urgency to fix from a direction I don't know about?

**Comment by cournape** [comment]
On my side, it is only about being able to update to latest pyparsing. I noticed this issue while updating pyparsing, and so put it back on the back burner.

**Comment by mdickinson** [comment]
@rkern's suggested fix implemented in #243.
